core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/DatalakeV4Endpoint.java

```diff
@@ -30,4 +34,33 @@ public interface DatalakeV4Endpoint {
     @ApiOperation(value = LIST_BY_WORKSPACE, produces = MediaType.APPLICATION_JSON, notes = Notes.STACK_NOTES, nickname = "listDatalakes")
     StackViewV4Responses list(@Deprecated @QueryParam("environment") String environment, @QueryParam("environmentCrn") String environmentCrn);
 
+    @GET
+    @Path("{platform}/diskTypeSupported")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Checks if Disk Type Change is supported", produces = MediaType.APPLICATION_JSON, nickname = "checksPlatformPermission")
+    Boolean isDiskTypeChangeSupported(@PathParam("platform") String platform);
+
+    @PUT
+    @Path("updateDisks")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Updates Disk Type and/ size", produces = MediaType.APPLICATION_JSON, nickname = "updateDiskTypeAndSize")
+    void updateDiskTypeAndSize(@Valid SdxDiskUpdateRequest sdxDiskUpdateRequest) throws Exception;
+
+    @PUT
+    @Path("{stackId}/stopCluster")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Stops CM services running on Datalake", produces = MediaType.APPLICATION_JSON, nickname = "stopCluster")
+    void stopCluster(@PathParam("stackId") long stackId) throws Exception;
```

Comment:
if this is about stoppping CM services then naming stopCluster is not corrent, stopCMServices should be used instead

Reply:
done. But even in ClusterModificationService (which is what I am calling here) it is defined as stop cluster - so i used the same

Reply:
it makes sense in ClusterModificationService because it is CM related, but here it is confusing, please do the rename